backend/infrahub/core/schema/schema_branch.py

```diff
@@ -226,8 +226,18 @@ def diff(self, other: SchemaBranch) -> SchemaDiff:
     def update(self, schema: SchemaBranch) -> None:
         """Update another SchemaBranch into this one."""
 
-        local_kinds = list(self.nodes.keys()) + list(self.generics.keys())
-        other_kinds = list(schema.nodes.keys()) + list(schema.generics.keys())
+        local_kinds = (
+            list(self.nodes.keys())
+            + list(self.generics.keys())
+            + list(self.profiles.keys())
+            + list(self.templates.keys())
+        )
+        other_kinds = (
+            list(schema.nodes.keys())
+            + list(schema.generics.keys())
+            + list(schema.profiles.keys())
+            + list(schema.templates.keys())
```

Comment:
Potentially we could have the listing of these keys as a private function, though perhaps mostly relevant if we have even more variations in the future.